publish/src/commands/deploy.js

```
@@ -1567,6 +1567,131 @@ const deploy = async ({
 		}
 	}
 
+	// Now do binary option market cache rebuilding
+	if (binaryOptionMarketManager) {
+		console.log(gray('Checking all binary option markets have rebuilt caches'));
+		let binaryOptionMarkets = [];
+		// now grab all possible binary option markets to rebuild caches as well
+		const binaryOptionsFetchPageSize = 100;
+		for (const marketType of ['Active', 'Matured']) {
+			const numBinaryOptionMarkets = Number(
+				await binaryOptionMarketManager.methods[`num${marketType}Markets`]().call()
+			);
+			console.log(
+				gray('Found'),
+				yellow(numBinaryOptionMarkets),
+				gray(marketType, 'binary option markets')
+			);
+
+			if (numBinaryOptionMarkets > binaryOptionsFetchPageSize) {
+				console.log(
+					redBright(
```

Show we throw here instead? Should probably never happen, but if it ever does, it'd be a shame to have it fail silently and not notice this red log.

---

Hmmmm. Probably better yeah.